### ISSUE TITLE
ar-dns integration

### ISSUE THREAD

**Issue by mcdermj** [mcdermj/buster#47]
The first stages of ar-dns.net integration are committed in f428fbba and 986199ac.  This is just a basic downloading of the preprocessed files that @johnhays provides.  There are a few things to note about the entire system.

1)  Buster's typedown functionality needs to have a complete list of at least reflector names so that it will work correctly.  This can't really happen via DNS without AXFRs, which would be a pain to implement compared to the relatively painless implementation currently.

2)  We may want to refresh those destination addresses periodically so that we get new information.  I'm not exactly sure how to make this happen.

I have thoughts about using the plist files just to seed the dropdown lists, and then not updating them at runtime.  Note with the current implementation, the new data is *not* saved to the disk.  I think this is the right thing to do.  Buster is inherently an Internet program, and I'm not going to pander to the edge cases where someone wants to use their ThumbDV non-Internet connected.

We could just refresh the information at some X time period and re-import the plist file.  This would be relatively easy to do with GCD and the various timers available.  We already do that with the various keepalives in the system, this wouldn't be that hard either.  That being said, I think the *right* thing to do would be to do the DNS lookup for the address at *link time* not at *startup time*.  When we get a link request, we *know* that the user intends to use the link, therefore refreshing the information is probably important.  If the repeater or reflector has changed addresses since launch time, we should at least try to reflect that.

Moving forward, I'd really like to encode more information in ar-dns.net about things.  Right now, I only really get name to DNS mappings and some crude idea of what protocols are supported.  What I'd kinda like is to use SRV and TXT records to convey some more data.  Maybe something like:

```
_dextra._udp.kf7ldg.ar-dns.net 86400 IN SRV 10 10 30001 kf7ldg.ar-dns.net.
_dplus._udp.kf7ldg.ar-dns.net  86400 IN SRV 10 10 20001 kf7ldg.ar-dns.net.
kf7ldg.ar-dns.net              86400 IN TXT "v=dstar1 proto_order=dplus,dextra,dcs"
```

Anyhow, these are ideas and I'd love to hear from @johnhays especially on what he things, but input might also be useful from @richark and @DonaldHays as well.

**Comment by johnhays** [comment]
Anna - I am more than happy to support srv and txt records on ar-dns.net, but would need support from the reflector/gateway operators.

**Comment by richark** [comment]
Just discovered edge case, ar-net.org is un-available, but Internet otherwise is accessible....

Today (2020-08-21) ar-net.org is down (at least from my QTH) and thus the list of reflectors IP addresses is very out of date.

**Comment by johnhays** [comment]
Yeah the server physically failed during the move. I have to check an old
server and see if I have code.

**Comment by mcdermj** [comment]
John,

If you’d like, I’m more than happy to pay for a list t3a.micro or something in AWS to run this. I’d surely give you an account with sudo. 

—Anna

Sent from my iPad

On Aug 21, 2020, at 10:01 PM, John Hays (K7VE) <notifications@github.com> wrote:

﻿


Yeah the server physically failed during the move. I have to check an old
server and see if I have code.

**Comment by richark** [comment]
I'll split the cost with Anna for the AWS instance....

I moved the WWARA site over and it was pretty painless.

Kenny

**Comment by johnhays** [comment]
May take you up on it. Have to get the old server up and see if I have
everything.

**Comment by bhhoyer** [comment]
NW Digital Radio would be happy to sponsor the server. What’s the cost?

Bryan

**Comment by G7FCT** [comment]
Is this issue still being looked into. Is there an easy way to manually add Refelctors etc to the list. ?

**Comment by johnhays** [comment]
Ar-dns is deprecated.

**Comment by G7FCT** [comment]
Thanks for the reply. Don't have any other options on the Mac which is a shame.
